core/include/cubos/core/ecs/component/reflection.hpp

```diff
@@ -0,0 +1,60 @@
+/// @file
+/// @brief Class @ref cubos::core::ecs::ComponentTypeBuilder.
+/// @ingroup core-ecs-component
+
+#pragma once
+
+#include <cubos/core/reflection/traits/constructible.hpp>
+#include <cubos/core/reflection/traits/fields.hpp>
+#include <cubos/core/reflection/type.hpp>
+
+namespace cubos::core::ecs
+{
+    /// @brief Builder for @ref reflection::Type objects which represent component types.
+    ///
+    /// Used to reduce the amount of boilerplate code required to define a component type.
+    /// Automatically adds the @ref reflection::ConstructibleTrait and @ref reflection::FieldsTrait.
+    /// The type @p T must be default-constructible, copy-constructible and move-constructible.
+    ///
+    /// @tparam T Component type.
+    template <typename T>
+    class ComponentTypeBuilder
+    {
+    public:
+        /// @brief Constructs.
+        /// @param name Component type name, including namespace.
+        ComponentTypeBuilder(std::string name)
+            : mType(reflection::Type::create(std::move(name)))
+        {
+            mType.with(reflection::ConstructibleTrait::typed<T>()
+                           .withDefaultConstructor()
+                           .withCopyConstructor()
+                           .withMoveConstructor()
```

Comment:
Adding these 3 constructors is so frequent we should have a short hand for them, maybe `.withBasicConstructors()`?